hindsight/output/crash_summary.lua

```diff
@@ -0,0 +1,52 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+Outputs a crash ping summary derived stream in the specified format one table/file per day.
+
+Config:
+
+filename = "crash_summary.lua"
+message_matcher = "Type == 'telemetry' && Fields[docType] == 'crash'"
+
+format      = "redshift.psv"
+buffer_path = "/mnt/output"
+buffer_size = 20 * 1024 * 1024
+s3_path     = "s3://test"
+
+--]]
+
+local ds = require "derived_stream"
+local fx = require "fx"
+local ping = require "fx.ping"
+
+local name = "crash_summary"
+local schema = {
+--  column name                 type            length  attributes  field /function
+    {"Timestamp"                ,"TIMESTAMP"    ,nil    ,"SORTKEY"  ,"Timestamp"},
+    {"crashDate"                ,"DATE"         ,nil    ,nil        ,function () return ping.get_date(ping.payload().payload.crashDate) end},
+    {"clientId"                 ,"CHAR"         ,36     ,"DISTKEY"  ,"Fields[clientId]"},
+    {"buildVersion"             ,"VARCHAR"      ,32     ,nil        ,function () return ping.build().version end},
```

Comment:
This data is consumed raw and it doesn't look like the postgres escaping makes it safe to insert.